src/main/java/com/thevoxelbox/voxelsniper/brush/PerformBrush.java

```diff
@@ -30,69 +30,67 @@
 
 import com.thevoxelbox.voxelsniper.Message;
 import com.thevoxelbox.voxelsniper.SnipeData;
+import com.thevoxelbox.voxelsniper.util.BlockHelper;
 import org.spongepowered.api.block.BlockState;
-import org.spongepowered.api.data.key.Key;
 import org.spongepowered.api.world.BlockChangeFlag;
 import org.spongepowered.api.world.BlockChangeFlags;
 import org.spongepowered.api.world.Location;
 import org.spongepowered.api.world.World;
 
 import java.util.Arrays;
 import java.util.Optional;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public abstract class PerformBrush extends Brush {
 
-    private static final Pattern PERFORMER = Pattern.compile("[mMcC][mMcC]?[pP]?");
+    private static final Pattern PERFORMER = Pattern.compile("([mic])([mic]?)(p?)");
 
-    protected PerformerType place = PerformerType.TYPE;
-    protected PerformerType replace = PerformerType.NONE;
-    protected boolean physics = true;
+    protected PerformerType placeMethod = PerformerType.TYPE;
+    protected PerformerType replaceMethod = PerformerType.NONE;
+    protected boolean usePhysics = true;
 
     public void parse(String[] args, SnipeData v) {
-        String handle = args[0];
-        if (PERFORMER.matcher(handle).matches()) {
-            char p = handle.charAt(0);
-            if (p == 'm' || p == 'M') {
-                this.place = PerformerType.TYPE;
-            } else if (p == 'c' || p == 'C') {
-                this.place = PerformerType.COMBO;
-            }
-            int i = 1;
-            if (handle.length() >= 2) {
-                char r = handle.charAt(i);
-                i = 2;
-                if (r == 'm' || r == 'M') {
-                    this.replace = PerformerType.TYPE;
-                } else if (r == 'c' || r == 'C') {
-                    this.replace = PerformerType.COMBO;
-                } else {
-                    i = 1;
-                    this.replace = PerformerType.NONE;
-                }
-            }
-            if (handle.length() >= i + 1) {
-                char e = handle.charAt(i);
-                if (e == 'p' || e == 'P') {
-                    this.physics = false;
-                } else {
-                    this.physics = true;
-                }
+        String handle = args[0].toLowerCase();
+        Matcher performersMatch = PERFORMER.matcher(handle);
+
+        if (performersMatch.matches()) {
+            placeMethod = stringToMethod(performersMatch.group(1));
+            if (placeMethod == PerformerType.NONE) {
+                throw new IllegalArgumentException("Unknown placement method '" + performersMatch.group(1) + "'");
             }
+
+            replaceMethod = stringToMethod(performersMatch.group(2));
```

Comment:
This should likewise inform the user if they've specified an invalid performer.

Reply:
At the moment, it wouldn't be possible to tell if the user has passed an invalid parameter or if they have actually passed in a brush argument.

Reply:
Ah yeah I see.